build/ci-build.yml

```
@@ -0,0 +1,122 @@
+name: Build Arcus.EventGrid CI
+
+trigger:
+  - master
+pr:
+  - master
+
+variables:
+  - group: 'Arcus Event Grid - Integration Testing'
+  - group: 'Arcus - GitHub Package Registry'
+  - group: 'Build Configuration'
+
+  # Always use fixed version for .NET Core SDK
+  - name: 'DotNet.Sdk.Version'
+    value: '2.2.105'
+  - name: 'Project'
+    value: 'Arcus.EventGrid'
+
+# Building the project:
+# - Download correct .NET Core SDK
+# - .NET Core build
+# - Publish build artifacts
+stages:
+  - stage: Build
+    jobs:
+      - job: Compile
+        pool:
+          vmImage: 'ubuntu-16.04'
+        steps:
+          - task: DotNetCoreInstaller@0
+            displayName: 'Import .NET Core SDK ($(DotNet.Sdk.Version))'
+            inputs:
+              version: '$(DotNet.Sdk.Version)'
+          - task: DotNetCoreCLI@2
+            displayName: 'Compile'
+            inputs:
+              projects: 'src/*.sln'
+              arguments: '--configuration $(Build.Configuration)'
+          - task: CopyFiles@2
+            displayName: 'Copy build artifacts'
+            inputs:
+              contents: '**/?(bin|obj)/**'
+              targetFolder: '$(Pipeline.Workspace)/build'
+          - task: PublishPipelineArtifact@0
+            displayName: 'Publish build artifacts'
+            inputs:
+              targetPath: '$(Pipeline.Workspace)/build'
+              artifactName: Build
+
+  # Testing unit tests of the project:
+  # - Download build artifacts
+  # - Download correct .NET Core SDK
+  # - .NET Core test
+  - stage: UnitTests
+    dependsOn: Build
+    condition: succeeded()
+    jobs:
+      - job: UnitTests
+        displayName: 'Run unit tests'
+        pool:
+          vmImage: 'ubuntu-16.04'
+        steps:
+          - task: DownloadPipelineArtifact@2
+            displayName: 'Download build artifacts'
+            inputs:
+              artifact: 'Build'
+              path: '$(Build.SourcesDirectory)'
+          - task: DotNetCoreInstaller@0
+            displayName: 'Import .NET Core SDK ($(DotNet.Sdk.Version))'
+            inputs:
+              version: '$(DotNet.Sdk.Version)'
+          - task: DotNetCoreCLI@2
+            displayName: 'Run unit tests'
+            inputs:
+              command: test
+              projects: 'src/**/$(Project).Tests.Unit.csproj'
+              arguments: '--configuration $(Build.Configuration)'
+              nobuild: true
+              publishTestResults: true
+
+  # Testing integration tests of the project:
+  # - Download build artifcats
+  # - Download correct .NET Core SDK
+  # - Replace tokens with project variabels
+  # - .NET Core test
+  - stage: IntegrationTests
+    dependsOn: Build
```

Let's add the following as well:
```suggestion
    displayName: Integration Tests
    dependsOn: Build
```